pdr_backend/lake/csv_data_store.py

```diff
@@ -0,0 +1,286 @@
+import os
+from typing import List, Optional
+import polars as pl
+
+from polars.type_aliases import SchemaDict
+
+
+class CSVDataStore:
+    def __init__(self, base_path: str):
+        self.base_path = base_path
+
+    def _get_folder_path(self, dataset_identifier: str) -> str:
+        """
+        Returns the folder path for the given dataset_identifier.
+        If the folder does not exist, it will be created.
+        @args:
+            dataset_identifier: str - identifier of the dataset
+        """
+        folder_path = os.path.join(self.base_path, dataset_identifier)
+        os.makedirs(folder_path, exist_ok=True)
+        return folder_path
+
+    def _fill_with_zero(self, number: int, length: int = 10) -> str:
+        """
+        Fills the given number with zeros to make it 10 digits long.
+        @args:
+            number: int - number to fill with zeros
+        """
+        number_str = str(number)
+        return f"{(length - len(number_str)) * '0'}{number_str}"
+
+    def _create_file_name(
+        self, dataset_identifier: str, start_time: int, end_time: int, row_count: int
+    ) -> str:
+        """
+        Creates a file name using the given dataset_identifier,
+        start_time, end_time, and row_count.
+        @args:
+            dataset_identifier: str - identifier of the dataset
+            start_time: int - start time of the data TIMESTAMP
+            end_time: int - end time of the data TIMESTAMP
+            row_count: int - number of rows in the data
+        """
+        start_time_str = self._fill_with_zero(start_time)
+        end_time_str = self._fill_with_zero(end_time)
+
+        return f"{dataset_identifier}_from_{start_time_str}_to_{end_time_str}_{row_count}.csv"
```

Comment:
Thank you for reviewing and discussing these details @kdetry

Something that I noted is that {row_count} helps us understand whether the file is append-able from the name, but it doesn't help with indexing or other features. Also, because it's a csv file, it's very quick to index/readline and get len().

I believe we can simplify it to just this. Preferably, we're simply appending to the end of a file...
```f"{dataset_identifier}_from_{start_time_str}_to_{end_time_str}"```

If you don't know when it ends, and it can be appended, you could tag it like so (rather than having an end_time_str that is changing all the time) so you know what an open-ended file looks like:
```f"{dataset_identifier}_from_{start_time_str}_to_"```

Reply:
Thank you for addressing this, and updating how files are saved.

This leads to a cleaner naming convention that is static (the name doesnt change after it's set), while letting you know which files can continue to be written to. Also, less logic.

We still have a problem where if we have +1000 events in a single timestamp, however this is a small concern at the moment.

Closing this feedback.